mapdataprefetcher/app/src/main/java/com/here/android/example/map/data/prefetcher/MapFragmentView.java

```diff
@@ -0,0 +1,440 @@
+
+    private AndroidXMapFragment m_mapFragment;
+    private Map m_map;
+
+    private MapDataPrefetcher m_mapDataPrefetcher;
+    private GeoCoordinate m_startNavPoint, m_endNavPoint;
+    private GeoBoundingBox m_geoBoundingBox;
+    private RoutePlan m_routePlan;
+    private Route m_route;
+    private NavigationManager m_navigationManager;
+
+    private ProgressBar m_progressBar;
+    private Button m_btnChangeMapState;
+
+    private MapType m_mapType;
+    private MapState m_mapState;
+    public boolean m_isRouteSimulationFinished = true;
+
+    private enum MapType {
+        MAP_BOUNDING_BOX,
+        ROAD_RADIUS
+    }
+
+    private enum MapState {
+        DEFAULT,
+        PREFETCH_MAP,
+        CALCULATE_ROUTE,
+        SIMULATE_NAVIGATION,
+        STOP_NAVIGATION
+    }
+
+    public MapFragmentView(AppCompatActivity activity) {
+        m_activity = activity;
+
+        initViews();
+        initMapFragment();
+    }
+
+    public boolean onCreateOptionsMenu(Menu menu) {
+        m_activity.getMenuInflater().inflate(R.menu.map_menu, menu);
+        return true;
+    }
+
+    public boolean onOptionsItemSelected(MenuItem item) {
+        if (m_isRouteSimulationFinished) {
+
+            if (!item.isChecked()) {
+                item.setChecked(true);
+
+                changeMapState(MapState.DEFAULT);
+
+                switch (item.getItemId()) {
+                    case R.id.mapBoundingBox:
+                        setMapType(MapType.MAP_BOUNDING_BOX);
+                        return true;
+                    case R.id.mapRoute:
+                        changeMapState(MapState.DEFAULT);
+
+                        setMapType(MapType.ROAD_RADIUS);
+                        return true;
+                }
+            }
+        } else {
+            Toast.makeText(m_activity, m_activity.getText(R.string.change_map_error),
+                    Toast.LENGTH_LONG).show();
+        }
+        return true;
+    }
+
+
+    private void loadMap() {
+        if (m_mapDataPrefetcher != null) {
+            m_map.removeAllMapObjects();
+            m_mapFragment.getPositionIndicator().setVisible(false);
+
+            MapEngine.setOnline(true);
+        }
+
+        if (m_mapType == MapType.MAP_BOUNDING_BOX) {
+            m_map.setZoomLevel(11.73);
+            m_map.setCenter(new GeoCoordinate(52.531003, 13.384783), Map.Animation.NONE);
+
+            changeMapState(MapState.PREFETCH_MAP);
+        } else {
+            showProgressBar(true);
+
+            showRoute();
+        }
+    }
+
+    private void setMapType(MapType mapType) {
+        m_mapType = mapType;
+
+        loadMap();
+    }
+
+    private AndroidXMapFragment getMapFragment() {
+        return (AndroidXMapFragment) m_activity.getSupportFragmentManager().findFragmentById(R.id.mapfragment);
+    }
+
+    private void initViews() {
+        m_progressBar = m_activity.findViewById(R.id.progress_circular);
+
+        showProgressBar(true);
```

Comment:
No need to show progress here. I think showing progress only makes sense during map prefetching, all other operations are quite fast( <1 sec).